sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -117,78 +123,217 @@ def hpat_pandas_series_getitem(self, idx):
 
     _func_name = 'Operator getitem().'
 
-    if not isinstance(self, SeriesType):
-        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
-
-    if isinstance(idx, types.Integer):
-        def hpat_pandas_series_getitem_idx_integer_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_series_iloc1
-            """
-
-            result = self._data[idx]
-            return result
-
-        return hpat_pandas_series_getitem_idx_integer_impl
-
+    if isinstance(self, SeriesOperatorTypeIloc):
```

Comment:
Could you please use `self._getitem_selector` variable instead classes like `SeriesOperatorTypeIloc`?

Reply:
Yes, I'll see how to do it.

Reply:
@shssf I made a selector. What do you think about this code?

Reply:
@1e-to I think it is more complicated than I thought. I didn't mean new class implementation. I thought about some small variable only which set by, for example, `iloc` method to the Series constructor and read by `getitem` implementation to distinguish these types.